Add Seiscomp scml0.14 compatibility, migrate code from sc3ml to scml

filefolder opened the pull request:
### What does this PR do?

Adds read/write capability for the new Seiscomp 7.0.0  `0.14` schema

There was a big shift on this release from using "sc3ml" to just "scml" which required a bit of juggling. Accordingly re-wrote a lot of code references from "sc3ml" to "scml" including adding the option to use `SCML` as a format option, keeping `SC3ML` in for backwards compatibility (should add a depreciation warning though).

Refreshes the published XSL files for all versions which usually is a great improvement on reading and writing. However there was a change/bugfix in their published XSL on 26.07.2024 regarding the confidence ellipsoids were calculated, seems they were off by 1000x due to a kilometer/meter mix-up. This was fixed from 0.10 on (which I believe was SC 3 or 4?) but not in 0.9 and below, so I had to skip testing on those. Some test data also had to be updated to suit. The older "wrong" versions of the XSL remain for legacy reasons (matching what is still included in SC) but I did drop 0.6 support as it was causing additional problems and is thoroughly decrepit. Users can always dig up the XLS files and do this manually if needed.

Another feature that I think was missing is the option to save SCML to whichever version you like. By default it is 0.12 which from memory is the last sort of significant update and should be compatible with versions 5+. 

I have added a few 0.14 tests but nothing that tests any of the more esoteric event types. As the changes from 0.13 are fairly small I don't envision anything weird happening.

I think important to have out for 1.5 ...

ThomasLecocq commented:
Solid work again @filefolder ! Thanks! and indeed, needed for 1.5.0.

We should get some seiscomp guys check this PR too:  @gempa-jabe @gempa-dirk could you have a look here ? It'd be cool to have your input ! 🥳

gempa-dirk reviewed (changes requested):
For style sheet conversion you may simply consider the files in https://github.com/SeisComP/common/blob/master/libs/xml/ in order to drop a line-by-line review.

gempa-dirk commented on obspy/io/seiscomp/inventory.py:
```diff
@@ -1000,7 +1000,7 @@ def _read_float_var(elem, cls, unit=False, datum=False, additional_mapping={}):
     tag names this is no longer possible; instead we just pass the value
     and not the tag name. We always set the unit/datum/uncertainties to None
-    because they are not provided by scml ?
+    because they are not provided by SCML ?
```
Question mark is confusing. Is this actually a question? Is so, it should be answered before providing code. See also above.

"provided by SCML" seems misleading. Either "provided by files in SCML" While, the statement is correct for unit, times in SCML do know value and uncertainty.

filefolder replied:
that was a legacy comment... agree about the ?s, have removed. 

not knowing how accurate this still is, I would change to: `We set the unit/datum/uncertainties to None because they are not present in SCML XSD.`

filefolder commented:
That was my intent but I am actually not too sure how to implement. Simple way could be (core/event/catalog.py and core/inventory/inventory.py) 

```
        if format == 'SC3ML':
            msg = "'SC3ML' will be depreciated soon, use 'SCML'"
            print(msg)
```
I am aware of `from obspy.core.util.deprecation_helpers import ObsPyDeprecationWarning` but I don't understand how or if it works.

filefolder commented:
OK, have rebased and added a deprecation warning (which actually manually changes format instances of `SC3ML` directly to `SCML` and removed it as an option)

```
cat = obspy.read_events('cat.scxml',format='SC3ML')
/usr/local/lib/python3.13/site-packages/obspy/core/util/decorator.py:209: ObsPyDeprecationWarning: Format 'SC3ML' is deprecated since ObsPy 1.5.0 and will be removed in a future release. Use 'SCML' instead.
  result = func(filename, *args, **kwargs)
```

gempa-dirk commented:
> For style sheet conversion you may simply consider the files in https://github.com/SeisComP/common/blob/master/libs/xml/ in order to drop a line-by-line review.

If  you have considered my previous recommendation, then I have no further comments.

filefolder commented on obspy/io/seiscomp/tests/data/iris_events.xml:
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<q:quakeml xmlns:q="http://quakeml.org/xmlns/quakeml/1.2" xmlns="http://quakeml.org/xmlns/bed/1.2">
```
good question-- there are three of this file. 

`./core/tests/data/iris_events.xml` & `./io/quakeml/tests/data/iris_events.xml` are identical but out of scope for this PR (todo?)

`./io/seiscomp/tests/data/iris_events.xml` is some specific SC test version that had to be converted/reconverted/washed so has the same info but structured differently.

megies replied:
Ah well, we can just leave like this for now :+1:
